### PR TITLE
Refactor 1

### PR DESCRIPTION
**Pull request by amandeeprikhi**
- Refactored the Auth service.
- Made separate files for each module.

### CODE REVIEW

File: auth/auth.js [hunk 1]
```diff
@@ -1,159 +1,112 @@
...
-			statusCode: 500,
-			statusText: 'Internal Server Error',
-			message: err.message
-		});
-	}
+  try {
+    let token = await utils.token.extractToken(req);
+
+    if (token) {
+      jwt.verify(token, SECRET, (err, decoded) => {
```
Comment (aakashvermaee) [inline auth/auth.js]: I think, this token verification code, should also be moved to token.js

File: auth/auth.js [hunk 2]
```diff
@@ -1,159 +1,112 @@
...
-		}
-	}
-}
-
-async function checkPassword(enteredPassword, storedPassword) {
-	let result = bcrypt.compareSync(enteredPassword, storedPassword);
-	return result;
-}
+const { SECRET } = require("../config/config");
+const utils = require("../utils/index");
```
Comment (aakashvermaee) [inline auth/auth.js]: Writing /index is optional but, anyway.

File: auth/auth.js (hunk 1 above)
Comment (aakashvermaee) [reply auth/auth.js]: You forgot to refactor this code part

Comment (amandeeprikhi) [reply auth/auth.js]: I am working on this.

Comment (amandeeprikhi) [reply auth/auth.js]: This is now complete

File: auth/auth.js (hunk 2 above)
Comment (amandeeprikhi) [reply auth/auth.js]: can i mark it as resolved for now?

File: auth/auth.js [hunk 3]
```diff
@@ -1,24 +1,12 @@
...
-            success: false,
-            message: "Token is not valid"
-          });
-        } else {
-          res.send({ decoded });
-          next();
-        }
-      });
+      let result = await utils.token.verifyToken(token);
+      res.send(result.decoded);
```
Comment (ankitchawla777) [inline auth/auth.js]: what if in case of error if jwt id not decoded then result.decoded will be undefined

File: utils/token.js [hunk 4]
```diff
@@ -31,23 +31,24 @@ exports.extractToken = async req => {
...
       }
-    })
+    });
     return result;
   } catch (err) {
     throw err;
   }
-}
\ No newline at end of file
+};
```
Comment (ankitchawla777) [inline utils/token.js]: async functions return promise either return a promise or dont make it async

**Comment by aakashvermaee** [comment]
@amandeeprikhi 
- Code Restructure take Ref from DB MSVC
- Add CORS

**Pull request merged**